modules/sdk-coin-sui/test/unit/transactionBuilder/stakingBuilder.ts

```diff
@@ -0,0 +1,126 @@
+import should from 'should';
+import { TransactionType } from '@bitgo/sdk-core';
+import utils from '../../../src/lib/utils';
+import { Transaction as SuiTransaction } from '../../../src/lib/transaction';
+import { MoveCallTx, SuiTransactionType } from '../../../src/lib/iface';
+
+describe('Sui Staking Builder', () => {
+  const factory = getBuilderFactory('tsui');
+
+  describe('Succeed', () => {
+    it('should build a staking addDelegation tx with one coin', async function () {
+      const txBuilder = factory.getStakingBuilder();
+      txBuilder.type(SuiTransactionType.AddDelegation);
+      txBuilder.sender(testData.STAKING_SENDER_ADDRESS);
+      txBuilder.requestAddDelegation(testData.requestAddDelegationTxOneCoin);
+
+      txBuilder.gasBudget(testData.STAKING_GAS_BUDGET);
+      txBuilder.gasPayment(testData.stakingGasPayment);
+      const tx = await txBuilder.build();
+      should.equal(tx.type, TransactionType.AddDelegator);
+      (tx as SuiTransaction<MoveCallTx>).suiTransaction.gasPayment.should.deepEqual(testData.stakingGasPayment);
+
+      tx.inputs.length.should.equal(1);
+      tx.inputs[0].should.deepEqual({
+        address: testData.STAKING_SENDER_ADDRESS,
+        value: testData.STAKING_AMOUNT.toString(),
+        coin: 'tsui',
+      });
+      tx.outputs.length.should.equal(1);
+      tx.outputs[0].should.deepEqual({
+        address: testData.VALIDATOR_ADDRESS,
+        value: testData.STAKING_AMOUNT.toString(),
+        coin: 'tsui',
+      });
+      const rawTx = tx.toBroadcastFormat();
+      should.equal(utils.isValidRawTransaction(rawTx), true);
+      should.equal(rawTx, testData.ADD_DELEGATION_TX_ONE_COIN);
+    });
+
+    it('should build a staking addDelegation tx with multiple coins', async function () {
+      const txBuilder = factory.getStakingBuilder();
+      txBuilder.type(SuiTransactionType.AddDelegation);
+      txBuilder.sender(testData.STAKING_SENDER_ADDRESS);
+      txBuilder.requestAddDelegation(testData.requestAddDelegationTxMultipleCoins);
+      txBuilder.gasBudget(testData.STAKING_GAS_BUDGET);
+      txBuilder.gasPayment(testData.stakingGasPayment);
+      const tx = await txBuilder.build();
+      should.equal(tx.type, TransactionType.AddDelegator);
+      (tx as SuiTransaction<MoveCallTx>).suiTransaction.gasPayment.should.deepEqual(testData.stakingGasPayment);
+
+      tx.inputs.length.should.equal(1);
+      tx.inputs[0].should.deepEqual({
+        address: testData.STAKING_SENDER_ADDRESS,
+        value: testData.STAKING_AMOUNT.toString(),
+        coin: 'tsui',
+      });
+      tx.outputs.length.should.equal(1);
+      tx.outputs[0].should.deepEqual({
+        address: testData.VALIDATOR_ADDRESS,
+        value: testData.STAKING_AMOUNT.toString(),
+        coin: 'tsui',
+      });
+      const rawTx = tx.toBroadcastFormat();
+      should.equal(utils.isValidRawTransaction(rawTx), true);
+      should.equal(rawTx, testData.ADD_DELEGATION_TX_MUL_COIN);
+    });
+  });
+
```

Comment:
can we add build from raw transaction.

Reply:
it's there `const fromRawTx = await factory.from(testData.ADD_DELEGATION_TX_MUL_COIN).build();` we can build from rawTx as well